src/storage/src/hummock/store/local_hummock_storage.rs

```diff
@@ -593,19 +593,21 @@ pub type LocalHummockStorageIterator<'a> = HummockStorageIteratorInner<'a>;
 
 pub struct HummockStorageIteratorInner<'a> {
     inner: UserIterator<HummockStorageIteratorPayloadInner<'a>>,
+    initial_read: bool,
     stats_guard: IterLocalMetricsGuard,
 }
 
 impl<'a> StateStoreIter for HummockStorageIteratorInner<'a> {
-    type Item = StateStoreIterItem;
-
-    async fn next(&mut self) -> StorageResult<Option<Self::Item>> {
+    async fn try_next<'b>(&'b mut self) -> StorageResult<Option<StateStoreIterItemRef<'b>>> {
         let iter = &mut self.inner;
+        if !self.initial_read {
+            self.initial_read = true;
+        } else {
+            iter.next().await?;
+        }
 
         if iter.is_valid() {
-            let kv = (iter.key().clone(), iter.value().clone());
-            iter.next().await?;
-            Ok(Some(kv))
+            Ok(Some((iter.key(), iter.value())))
```

Comment:
When will `inital_read` be used? It seems the behavior of the iterator is not modified.

Reply:
This is the key difference between the behavior of hummock iter and state store iter. The hummock iter can be seen as calling the first `next` in the `rewind` or `seek`. Therefore, in the first `next` call of state store iter, we should not call hummock_iter.next(), and the `initial_read` is used to track whether the current `next` call is the first one or not.